lib/logflare_web/controllers/endpoints_controller.ex

```diff
@@ -16,16 +19,32 @@ defmodule LogflareWeb.EndpointsController do
     methods: ["GET", "POST", "OPTIONS"],
     send_preflight_response?: true
 
-  def query(conn, %{"token" => token}) do
+  def query(%{assigns: assigns} = conn, %{"token" => token}) do
     endpoint_query = Endpoints.get_mapped_query_by_token(token)
+    user = Map.get(assigns, :user)
 
-    case Endpoints.run_cached_query(endpoint_query, conn.query_params) do
-      {:ok, result} ->
-        Logger.debug("Endpoint cache result, #{inspect(result, pretty: true)}")
-        render(conn, "query.json", result: result.rows)
+    with :ok <- check_auth(endpoint_query, user),
```

Comment:
Is there a reason you didn't make a plug out of this? 

With a plug you don't have to worry about this logic in the controller, and it's what we do in all the other controllers unless we have to do something otherwise.

Reply:
reason why i moved down the resource ownership check to controller level was because the VerifyApiAccess was doing a token checks and didn't feel right to do ownership checks in there. I can move up the ownership check to a separate plug, and cover sources and controller as well.